src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -2010,6 +2011,223 @@ public void shouldRetrieveItemsViaPost() throws InterruptedException, ExecutionE
     assertThat(responseJson.getInteger("totalRecords"), is(numOfItemsToCreate));
   }
 
+  @Test
+  @SneakyThrows
+  public void shouldUpdateItems_positive() {
+    var holdingsRecordId = createInstanceAndHolding(MAIN_LIBRARY_LOCATION_ID);
+    var itemId = randomUUID();
+    // create item
+    createItem(smallAngryPlanet(itemId, holdingsRecordId));
+
+    // get item
+    CompletableFuture<Response> completableFuture = new CompletableFuture<>();
+    getClient().get(itemsStorageUrl("/" + itemId), TENANT_ID,
+      ResponseHandler.any(completableFuture));
+    var response = completableFuture.get(TIMEOUT, TimeUnit.SECONDS);
+    assertThat(response.getStatusCode(), is(200));
+
+    // modify item body
+    var itemsJson = response.getJson();
+    itemsJson.remove("hrid");
+    itemsJson.put("barcode", null);
+    itemsJson.put("order", 55);
+
+    // update item via PATCH
+    CompletableFuture<Response> patchCompleted = new CompletableFuture<>();
+    getClient().patch(itemsStorageUrl(""), new JsonObject().put("items", new JsonArray().add(itemsJson)),
+      TENANT_ID, ResponseHandler.empty(patchCompleted));
+    var patchResponse = patchCompleted.get(TIMEOUT, TimeUnit.SECONDS);
+    assertThat(patchResponse.getStatusCode(), is(204));
+
+    // verify item is updated
+    CompletableFuture<Response> itemFuture = new CompletableFuture<>();
+    getClient().get(itemsStorageUrl("/" + itemId), TENANT_ID,
+      ResponseHandler.json(itemFuture));
+    var itemResponse = itemFuture.get(TIMEOUT, TimeUnit.SECONDS);
```

Comment:
Couldn't we use `getById(itemId)` here and remove additional call from message publish check?